### PR TITLE
ISPN-12800 Add some missing JVM metrics

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12800

### CODE REVIEW

File: server/rest/src/main/resources/io/smallrye/metrics/base-metrics.properties [hunk 1]
```diff
@@ -82,6 +82,16 @@ memory.usedNonHeap.displayName: Used Non Heap Memory
 memory.usedNonHeap.description: Displays the amount of used memory.
 memory.usedNonHeap.type: gauge
 memory.usedNonHeap.unit: bytes
+memory.initHeap.mbean: java.lang:type=Memory/HeapMemoryUsage#init
+memory.initHeap.displayName: Init Heap Memory
+memory.initHeap.description: Displays the initial amount of allocated memory in bytes for the heap.
```
Comment (oraNod) [inline server/rest/src/main/resources/io/smallrye/metrics/base-metrics.properties:87]:
```suggestion
memory.initHeap.description: Displays the initial amount of allocated heap memory in bytes.
```

File: server/rest/src/main/resources/io/smallrye/metrics/base-metrics.properties [hunk 2]
```diff
@@ -82,6 +82,16 @@ memory.usedNonHeap.displayName: Used Non Heap Memory
...
+memory.initHeap.unit: bytes
+memory.initHeap.type: gauge
+memory.initNonHeap.mbean: java.lang:type=Memory/NonHeapMemoryUsage#init
+memory.initNonHeap.displayName: Init Non-Heap Memory
+memory.initNonHeap.description: Displays the initial amount of allocated memory in bytes for non-heap.
```
Comment (oraNod) [inline server/rest/src/main/resources/io/smallrye/metrics/base-metrics.properties:92]:
```suggestion
memory.initNonHeap.description: Displays the initial amount of allocated off-heap memory in bytes.
```

**Comment by oraNod** [comment]
Two minor suggestions but LGTM otherwise. I can see these two gauges from the metrics endpoint.

**Comment by gustavonalle** [comment]
As a side note, there are tons of JVM metrics already exposed via REST https://infinispan.org/docs/stable/titles/rest/rest.html#rest_v2_server_memory

**Comment by tristantarrant** [comment]
@gustavonalle yes, but the request here was for these metrics to be available through Prometheus

**Comment by anistor** [comment]
LGTM, if the wording was updated.

**Comment by tristantarrant** [comment]
I've squashed the commits

**Comment by anistor** [comment]
Merged in master. Thanks !

**Pull request merged**